homeassistant/components/imap/coordinator.py

```diff
@@ -165,14 +165,12 @@ def _decode_payload(part: Message) -> str:
             Falls back to the raw content part if decoding fails.
             """
             try:
-                content_charset = part.get_content_charset()
-                decoded_payload: bytes = part.get_payload(decode=True)
-                return (
-                    decoded_payload.decode(content_charset)
-                    if content_charset is not None
-                    else str(decoded_payload)
-                )
-            except ValueError:
+                decoded_payload: Any = part.get_payload(decode=True)
+                assert isinstance(decoded_payload, bytes)
```

Comment:
Assert statements can be turned off. We can't use it in production.

Reply:
Added `TYPE_CHECKING` condition.